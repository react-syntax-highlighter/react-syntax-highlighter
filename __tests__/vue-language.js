import React from 'react';
import renderer from 'react-test-renderer';
import SyntaxHighlighter from '../src';

const code = `<script lang="ts" setup>
import { Layout } from "@ui/libs";
</script>

<template>
    <Layout />
</template>
`;

test('SyntaxHighlighter component renders vue', () => {
  const tree = renderer
    .create(<SyntaxHighlighter language="vue">{code}</SyntaxHighlighter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
