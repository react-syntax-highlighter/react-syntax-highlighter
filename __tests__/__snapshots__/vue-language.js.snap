// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyntaxHighlighter component renders vue 1`] = `
<pre
  style={
    Object {
      "background": "#F0F0F0",
      "color": "#444",
      "display": "block",
      "overflowX": "auto",
      "padding": "0.5em",
    }
  }
>
  <code
    className="language-vue"
    style={
      Object {
        "whiteSpace": "pre",
      }
    }
  >
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &lt;
    </span>
    <span
      className="xml hljs-tag"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      script
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
       
    </span>
    <span
      className="xml hljs-tag hljs-attr"
      style={Object {}}
    >
      lang
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      =
    </span>
    <span
      className="xml hljs-tag"
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      "ts"
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
       
    </span>
    <span
      className="xml hljs-tag hljs-attr"
      style={Object {}}
    >
      setup
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &gt;
    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
      

    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
      
    </span>
    <span
      className="xml javascript"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      import
    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
       
    </span>
    <span
      style={
        Object {
          "color": "#BC6060",
        }
      }
    >
      { Layout }
    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
       
    </span>
    <span
      className="xml javascript"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      from
    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
       
    </span>
    <span
      className="xml javascript"
      style={
        Object {
          "color": "#880000",
        }
      }
    >
      "@ui/libs"
    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
      ;

    </span>
    <span
      className="xml javascript"
      style={Object {}}
    >
      
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &lt;/
    </span>
    <span
      className="xml hljs-tag"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      script
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &gt;
    </span>
    <span
      className="xml"
      style={Object {}}
    >
      

    </span>
    <span
      className="xml"
      style={Object {}}
    >
      

    </span>
    <span
      className="xml"
      style={Object {}}
    >
      
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &lt;
    </span>
    <span
      className="xml hljs-tag"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      template
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &gt;
    </span>
    <span
      className="xml"
      style={Object {}}
    >
      

    </span>
    <span
      className="xml"
      style={Object {}}
    >
          
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &lt;
    </span>
    <span
      className="xml hljs-tag"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Layout
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
       /&gt;
    </span>
    <span
      className="xml"
      style={Object {}}
    >
      

    </span>
    <span
      className="xml"
      style={Object {}}
    >
      
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &lt;/
    </span>
    <span
      className="xml hljs-tag"
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      template
    </span>
    <span
      className="xml hljs-tag"
      style={Object {}}
    >
      &gt;
    </span>
    <span
      className="xml"
      style={Object {}}
    >
      

    </span>
    <span
      className="xml"
      style={Object {}}
    >
      
    </span>
  </code>
</pre>
`;
